enhancements/worker-latency-profile/worker-latency-profile.md

```diff
@@ -0,0 +1,450 @@
+
+## Motivation
+
+To accommodate the higher network latency (such as Edge cases) cluster may experience, we need to adjust how frequently the `Kubelet` updates its status and how components such as `Kube Controller Manager` and `Kube API Server`react to those status updates.
+
+The main motivation of this enhancement is to allow setting relevant arguments for these critical components in a more controlled manner instead of letting the users directly modify them manually which will make the cluster unsupported. This will also eliminate any room for manual errors that could lead to unscheduled downtime.
+
+### Goals
+
+* Enable Openshift cluster to fine tune the reliability in medium to high latency scenarios by specifying simple `WorkerLatencyProfile` at day-0 as well as day-2.
+
+### Non-Goals
+
+* Modify the existing `Kubelet`, `Kube Controller Manager` or `Kube API Server` code in any way.
+* Allow any kind of different latency scenarios between masters
+
+### User Stories
+
+* User wants to fine tune the cluster reliability for their node latency scenario while making sure they are protected from setting parameters that could potentially break the cluster.
+
+## Proposal
+
+* The option to set `WorkerLatencyProfile` will have to reside in a centralized location. The [OpenShift Infrastructure config object](https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L28) contains information describing how a cluster functions including cloud config and platform specification for each cloud. Setting the  `WorkerLatencyProfile` is an infrastructure setting.
+* [MCO](https://github.com/openshift/machine-config-operator) set the appropriate value of the `Kubelet` flag `--node-status-update-frequency`
+* [KCMO](https://github.com/openshift/cluster-kube-controller-manager-operator) set the appropriate value of the `Kube Controller Manager` flag `--node-monitor-grace-period`
+* [CKAO](https://github.com/openshift/cluster-kube-apiserver-operator)set the appropriate value of the `Kube API Server` flags `--default-not-ready-toleration-seconds` and `--default-unreachable-toleration-seconds`
+
+## Design Details
+
+### API Extensions
+
+```go
+type WorkerLatencyProfileType string
+
+const (
+    // Medium Update and Average Reaction
+    MediumUpdateAndAverageReaction WorkerLatencyProfileType = "MediumUpdateAndAverageReaction"
+
+    // Low Update and Slow Reaction
+    LowUpdateAndSlowReaction WorkerLatencyProfileType = "LowUpdateAndSlowReaction"
+
+    // Default values of relavent Kubelet, Kube Controller Manager and Kube API Server
+    Default WorkerLatencyProfileType = "Default"
+)
+
+type Node struct {
```

Comment:
This code has too many information, focus only on new additions, so it's more readable.

Reply:
I removed the declaration of `ConditionStatus` and `ObjectReference` because they are generic enough. But I am afraid  dropping anything more than would lead to loss of information that could be helpful during implementation.